src/util.c

```diff
@@ -385,3 +384,23 @@ parse_key_value_string (char *kv_str,
 out:
     return rc;
 }
+
+#define ENTROPY_SRC "/dev/urandom"
+static void* ptr_mask = NULL;
+
+void
+util_init (void)
+{
+    g_assert (ptr_mask == NULL);
+    Random *rand = random_new ();
+    g_assert (random_seed_from_file (rand, ENTROPY_SRC) != -1);
+    g_assert (random_get_bytes (rand, (uint8_t*)&ptr_mask, sizeof (ptr_mask)) == sizeof (ptr_mask));
+    g_clear_object (&rand);
```

Comment:
Wouldn't you want to try again got a random number while ptr_mask is NULL?

Reply:
Yeah so why have it fail late when you can prevent it upfront?

Reply:
I'd rather the code fail after getting 4 bytes of `0`s out of `/dev/urandom` since this is an exceptional circumstance. I don't object to retrying some fixed number of times but if `/dev/urandom` already gave us 4 consecutive bytes of `0` I don't know that it's going to give us anything useful.

Reply:
I'd at least try a fixed number of times and fail in initialization. I'm assuming those asserts turn into abort() calls on debug builds.